src/v2/Apps/Search2/routes/Search2Results/Search2Results.tsx

```diff
@@ -0,0 +1,128 @@
+import { useState, useMemo, useRef, useEffect } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { Search2Results_system } from "v2/__generated__/Search2Results_system.graphql"
+import algoliasearch from "algoliasearch"
+import {
+  InstantSearch,
+  Configure,
+  connectPagination,
+  connectStateResults,
+} from "react-instantsearch-core"
```

Comment:
**praise:** Ahh, nice. So these were all just re-exported by `react-instantsearch-dom`, which is why we're able to instead use these directly from `react-instantsearch-core` if we're creating our own DOM, yes?

Reply:
Yes, you are absolutely right. After looking at [the source code](https://github.com/algolia/react-instantsearch/blob/master/packages/react-instantsearch-dom/src/index.js#L14-L43) of `react-instantsearch-dom`, you can see that all the connectors we need are imported from `react-instantsearch-core`. And for this reason, we can safely abandon the use of `react-instantsearch-dom`

Reply:
What exactly does this package provide? It's not immediately clear to me?

Reply:
`react-instantsearch-core ` contains all the basic logic for working with Algolia in React

Reply:
But like: what's that actually mean? We're just hitting Metaphysics ultimately, right?

Reply:
> We're just hitting Metaphysics ultimately, right?

Not really. Metaphysics on the new search page is only [needed](https://github.com/artsy/force/blob/02f7229e1a7b523d1cdf6bef8314c4823626769b/src/v2/Apps/Search2/routes/Search2Results/Search2Results.tsx#L45) to get the app id and api key for Algolia. All search results will be displayed **from Algolia**

Reply:
For example, for "Artist" pill the results from [this Algolia index](https://www.algolia.com/apps/CQWT6HEHVH/explorer/browse/Artist_staging?query=banksy&searchMode=search) will be displayed

Reply:
Ah, OK. I continue to think it is a very bad idea to not encapsulate an external service within Metaphysics — essentially working around the entire problem it solves... but I'm sure your team has thought through all the problems if it's gotten to this point.

Reply:
> Ah, OK. I continue to think it is a very bad idea to not encapsulate an external service within Metaphysics — essentially working around the entire problem it solves... but I'm sure your team has thought through all the problems if it's gotten to this point.

I think the most compelling reasons _not_ to encapsulate and abstract Algolia within Metaphysics have been:
- To support Algolia's automatic routing of the search request to the closest data center - minimizing latency
- Not needing to spend time building out a potentially complex proxying layer

Definitely not too late to experiment with other approaches though. Do you think we should explore using the InstantSearch library with a custom backend proxy and/or expose a new GraphQL field that abstracts most of the Algolia bits away? 

For example, we could explore an approach that would:
- Expose a new API endpoint in Gravity that proxies Algolia-formatted requests to Algolia
- Expose new Metaphysics fields to power searching/filtering via Algolia

https://www.algolia.com/doc/guides/building-search-ui/going-further/backend-search/in-depth/backend-instantsearch/js/#searching-for-results
https://github.com/charlypoly/algolia-graphql-schema

During very early [conversations](https://www.notion.so/artsy/Integration-with-Search-Vendor-2bb4f7dd196a456394f26477d97d4b58) - there was quite a bit of concern against maintaining too much of our own search infrastructure but perhaps we swung too far in that direction. We can still successfully leverage a backend system like Algolia while owning more of the glue that connects it with our apps via Metaphysics or other APIs.

Reply:
I'm mostly concerned as to how this plays with the existing schema — if Algolia makes it into all filtered artwork sets then the vast majority of the UI depends on it. I guess I just don't understand the mechanics of how we would use our existing UI and fragments inside of Algolia without just simply hitting Metaphysics again anyway.

How much performance really are we actually saving by hitting Algolia directly instead of stitching it into Metaphysics?